torchvision/prototype/datasets/utils/_dataset.py

```diff
@@ -78,6 +78,15 @@ def read_categories_file(path: pathlib.Path) -> List[List[str]]:
             return [row for row in csv.reader(file)]
 
     def make_config(self, **options: Any) -> DatasetConfig:
+        if not self._valid_options:
+            if options:
+                raise ValueError(
+                    f"Dataset {self.name} does not take any options, "
+                    f"but got {sequence_to_str(list(options), separate_last=' and')}."
+                )
+
+            return DatasetConfig()
```

Comment:
Would `return DatasetConfig(self.default_config, **options)` work here?


Because if it does we could simplify to just

```py
        if not self._valid_options and options:
                raise ValueError(
                    f"Dataset {self.name} does not take any options, "
                    f"but got {sequence_to_str(list(options), separate_last=' and')}."
                )
```

and return only at the end

Reply:
Yes it would. Both `self.default_config` and `options` would be empty dicts here. I thought an early exit if no options are available and none are specified would be preferable, but I don't mind your suggestion either.